samples/drivers/display/README.rst

```diff
@@ -0,0 +1,47 @@
+.. _display-sample:
+
+Display Sample
+##############
+
+Overview
+********
+
+This sample will draw some basic rectangles onto the display.
+The rectangle colors and positions are chosen so that you can check the
+orientation of the LCD and correct RGB bit order. The rectangles are drawn
+in clockwise order, from top left corner: red, green, blue, grey. The shade of
+grey changes from black through to white. If the grey looks too green or red
+at any point then the LCD may be endian swapped.
```

Comment:
I had to byte swap `fill_buffer_rgb565` to get it to look right on mimxrt1050_evk

Reply:
Sound like an endian issue mcux_elcdif driver, we could add bgr565 as pixel format, update the API header file to specify that the pixel formats are byte-order and return bgr565 from the mcux_elcdif driver.

Reply:
That's funny.  My previous comments ask what someone should do if indeed the LCD is endian swapped and @MaureenHelm hit this.  So maybe this is a common enough problem to say more here about what the developer might need to do in this situation?

Reply:
@dbkinder Problem is that there is not a common solution and it just a bug in the display driver.

Reply:
I'm not convinced it's a bug in the driver. If that were the case, wouldn't lvgl also not work? It looks like the sample is assuming a particular endianness with:
```c
	for (size_t idx = 0; idx < buf_size; idx += 2) {
		*(buf + idx + 0) = (color >> 8) & 0xFFu;
		*(buf + idx + 1) = (color >> 0) & 0xFFu;
	}
```

Reply:
> wouldn't lvgl also not work?

Yes if they would use RGB565 byte ordering but their code uses BGR565

> It looks like the sample is assuming a particular endianness

Actually the API is assuming this, if the code is changed it will break the SPI based display shields

Reply:
Agreed. Per your earlier suggestion, I think we need to add BGR565 as a pixel format and clarify the byte order in the API.

Reply:
@MaureenHelm Ok I will take care of it, is it ok I do this in a separate PR?

Reply:
Yes, thank you